File: conformance/impls/python/driver.py
#!/usr/bin/env python3
"""Conformance driver for the Python implementation. See ../../protocol.md."""

import json
import os
import sys

sys.path.insert(0, os.path.join(os.path.dirname(__file__), "..", "..", "..", "python"))

case = json.load(open(sys.argv[1]))
for k, v in (case.get("env") or {}).items():
    os.environ[k] = v

from unilog import _record, _resource  # noqa: E402 (env must be set first)

opts = case["input"]
rec = _record.build(
    severity_number=opts.get("severityNumber", 9),
    body=opts.get("body", ""),
    event_name=opts.get("eventName"),
    resource=_resource.get(),
    attributes=opts.get("attributes"),
    trace_id=opts.get("traceId"),
    span_id=opts.get("spanId"),
)
sys.stdout.write(_record.to_json_line(rec))

File: conformance/impls/php/driver.php
<?php

declare(strict_types=1);

// Conformance driver for the PHP implementation. See ../../protocol.md.

$case = json_decode(file_get_contents($argv[1]), true, flags: JSON_THROW_ON_ERROR);
foreach ($case['env'] ?? [] as $k => $v) {
    putenv("{$k}={$v}");
}

require __DIR__ . '/../../../php/src/Severity.php';
require __DIR__ . '/../../../php/src/Redactor.php';
require __DIR__ . '/../../../php/src/Resource.php';
require __DIR__ . '/../../../php/src/Record.php';

$opts = $case['input'];
$record = \Unilog\Record::build([
    'severityNumber' => $opts['severityNumber'] ?? 9,
    'body' => $opts['body'] ?? '',
    'eventName' => $opts['eventName'] ?? null,
    'resource' => \Unilog\Resource::get(),
    'attributes' => $opts['attributes'] ?? [],
    'traceId' => $opts['traceId'] ?? null,
    'spanId' => $opts['spanId'] ?? null,
]);
echo \Unilog\Record::toJsonLine($record);
